File: matrixmult/MatrixMult.c
#include <stdlib.h>
#include <stdio.h>
#include <time.h>

int main(int argc, char** argv)
{
    //initilaization
    int n = 800;
		
    int A[n][n];
    int B[n][n];
    int C[n][n];

    srand(time(NULL));

    // populate matrices with random numbers from 0-99
    for(int i=0; i<n; i++)
            for(int j=0; j<n; j++)
                A[i][j] = rand() % 99;     
			
    for(int i=0; i<n; i++)
        for(int j=0; j<n; j++)
            B[i][j] = rand() % 99;

    // fill matrix c with zeroes
    for(int i=0; i<n; i++)
        for(int j=0; j<n; j++)
            C[i][j] = 0;


    clock_t start = clock(), diff;
    
    // perform the multiplication
    for(int i=0; i<n; i++)
    {
        for(int j=0; j<n; j++)
        {
            for(int k=0; k<n; k++)
                C[i][j] += A[i][k]*B[k][j];
        }
    }

    diff = clock() - start;
    int millisec = diff * 1000 / CLOCKS_PER_SEC; 

    printf("%d milliseconds\n", millisec);
    return 0;
}

File: matrixmult/MatrixMult.py
import numpy as np
import random
import time

# initialization
n = 800

A = np.reshape(np.array([0] * n**2, dtype = int), (n, n))
B = np.reshape(np.array([0] * n**2, dtype = int), (n, n))
C = np.reshape(np.array([0] * n**2, dtype = int), (n, n))

# populate matrices with int values from 0 to 99
for i in range(n):
    for j in range(n):
        A[i][j] = random.randint(0, 5)
                
for i in range(n):
    for j in range(n):
        B[i][j] = random.randint(0, 5)

#start timer here
start = time.time() * 1000

#perform matrix multiplication
for i in range(n):
    for j in range(n):
        C[i][j] = sum(A[i,:] * B[:,j])
        
end = time.time() * 1000

print(f'{int(end - start)}', 'milliseconds')